WpfTest/ViewModel/LoginViewModel.cs

```diff
@@ -0,0 +1,118 @@
+            RedirectToRegisterPage = new RelayCommand(Redirect);
+        }
+
+        public LoginModel LoginModel
+        {
+            get => _loginModel;
+            set
+            {
+                _loginModel = value;
+                OnPropertyChanged(nameof(LoginModel));
+            }
+        }
+
+        public string LoginError
+        {
+            get => _loginError;
+            set
+            {
+                _loginError = value;
+                OnPropertyChanged(nameof(LoginError));
+            }
+        }
+
+        public bool IsLoginSuccessful
+        {
+            get => _isLoginSuccessful;
+            set
+            {
+                _isLoginSuccessful = value;
+                OnPropertyChanged(nameof(IsLoginSuccessful));
+            }
+        }
+        
+        public ICommand LoginCommand { get; }
+        public ICommand RedirectToRegisterPage { get; }
+
+        private void OnLogin(object parameter)
+        {
+            ValidationResult results = _loginValidator.Validate(LoginModel);
+            if (results.IsValid)
+            {
+                // Simulating a login check
+                if (_loginModel.Login == "admin" && _loginModel.Password == "admin")
+                {
+                    IsLoginSuccessful = true;
+                    RedirectToMainWindow();
+                }
+                else
+                {
+                    IsLoginSuccessful = false;
+                    LoginError = "Invalid username or password.";
+                }
+            }
+            else
+            {
+                IsLoginSuccessful = false;
+                LoginError = string.Join(Environment.NewLine, results.Errors.Select(e => e.ErrorMessage));
+            }
+        }
+
+        private bool CanLogin(object parameter)
+        {
+            return !string.IsNullOrEmpty(_loginModel.Login) && !string.IsNullOrEmpty(_loginModel.Password);
+        }
+
+        private void RedirectToMainWindow()
+        {
+            MainPage mainPage = new MainPage();
+            mainPage.Show();
+        }
+
+        private void Redirect(object parameter)
+        {
+            RegisterView registerView = new RegisterView();
+            registerView.Show();
+        }
+
+        public string this[string columnName]
+        {
+            get
+            {
+                var firstOrDefault = _loginValidator.Validate(_loginModel).Errors.FirstOrDefault(error => error.PropertyName == columnName);
+                return firstOrDefault != null ? firstOrDefault.ErrorMessage : string.Empty;
+            }
+        }
+
+        public string Error
+        {
+            get
+            {
+                var results = _loginValidator.Validate(_loginModel);
+                return results != null && results.Errors.Any() ? string.Join(Environment.NewLine, results.Errors.Select(x => x.ErrorMessage).ToArray()) : string.Empty;
+            }
+        }
```

Comment:
I didn't understand this code, why you duplicate this logic from 100 line? 

For example in metanit https://metanit.com/sharp/wpf/14.php they just use indexator message and Error as just `NotImplementedException`.

probably in your code you make the same validation twice on the same object, I like this approach with `IDataErrorInfo`, but please consider this functionality more detailed what should be in Error and in indexator.